packages/extras/src/lib/hooks/useAsciiEffect.svelte.ts

```diff
@@ -0,0 +1,85 @@
+import { useThrelte } from '@threlte/core'
+import { AsciiEffect } from 'three/examples/jsm/effects/AsciiEffect.js'
+import type { AsciiEffectOptions } from 'three/examples/jsm/effects/AsciiEffect.js'
+import type { Camera, Scene, WebGLRenderer } from 'three'
+import { fromStore } from 'svelte/store'
+import { getContext, setContext } from 'svelte'
+
+const KEY = {}
+
+export const useAsciiEffect = (
+  getters: Partial<{
+    getRenderer: () => WebGLRenderer
+    getCharSet: () => string
+    getOptions: () => AsciiEffectOptions
+    getForegroundColor: () => string
+    getBackgroundColor: () => string
+  }> = {}
+) => {
+  const { camera: threlteCamera, canvas, dom, renderer, scene: threlteScene, size } = useThrelte()
+
+  const {
+    getRenderer = () => renderer,
+    getCharSet = () => '',
+    getOptions = () => ({}),
+    getForegroundColor = () => 'white',
+    getBackgroundColor = () => 'black'
+  } = getters
+
+  // save a ref in case it is used in a render loop
+  let ref: AsciiEffect
+  const asciiEffect = $derived.by(() => {
+    const effect = new AsciiEffect(getRenderer(), getCharSet(), getOptions())
+
+    effect.domElement.style.position = 'absolute'
+    effect.domElement.style.top = '0px'
+    effect.domElement.style.left = '0px'
+    effect.domElement.style.pointerEvents = 'none'
+
+    return (ref = effect)
+  })
+
+  const size$ = fromStore(size)
```

Comment:
`size$` is a bit confusing since it's not a svelte convention, I'd just go with
```ts
  const size = fromStore(sizeStore) // Renamed in useThrelte call
```